hepstats/hypotests/calculators/frequentist_calculator.py

```diff
@@ -0,0 +1,425 @@
+
+                >>> obs = zfit.Space('x', limits=(0.1, 2.0))
+                >>> data = zfit.data.Data.from_numpy(obs=obs, array=np.random.normal(1.2, 0.1, 10000))
+                >>> mean = zfit.Parameter("mu", 1.2)
+                >>> sigma = zfit.Parameter("sigma", 0.1)
+                >>> model = zfit.pdf.Gauss(obs=obs, mu=mean, sigma=sigma)
+                >>> loss = UnbinnedNLL(model=[model], data=[data], fit_range=[obs])
+
+                >>> calc = FrequentistCalculator(input=loss, minimizer=MinuitMinimizer(), ntoysnull=1000, ntoysalt=1000)
+        """
+
+        super(FrequentistCalculator, self).__init__(input, minimizer)
+
+        self._toysnull = {}
+        self._toysalt = {}
+        self._ntoysnull = ntoysnull
+        self._ntoysalt = ntoysalt
+        self._sampler = sampler
+        self._sample = sample
+        self._toys_loss = {}
+
+    @property
+    def ntoysnull(self):
+        """
+        Returns the number of toys loss for the null hypothesis.
+        """
+        return self._ntoysnull
+
+    @property
+    def ntoysalt(self):
+        """
+        Returns the number of toys loss for the alternative hypothesis.
+        """
+        return self._ntoysalt
+
+    def sampler(self, floating_params=None):
+        """
+        Create sampler with models.
+
+        Args:
+            floating_params (list): floating parameters in the sampler
+
+        Example with `zfit`:
+            >>> sampler = calc.sampler(floating_params=[zfit.Parameter("mean")])
+        """
+        self.set_dependents_to_bestfit()
+        nevents = []
+        # self.model and self.data defined in BaseCalculator
+        for m, d in zip(self.model, self.data):
+            if m.is_extended:
+                nevents.append("extended")
+            else:
+                nevents.append(get_nevents(d))
+
+        return self._sampler(self.model,  nevents, floating_params)
+
+    def sample(self, sampler, ntoys, parameter=None, value=None):
+        """
+        Returns the samples generated from the sampler for a given value of a parameter
+
+        Args:
+            sampler (list): generator of samples
+            ntoys (int): number of samples to generate
+            parameter (optional): floating parameter in the sampler
+            value (optional): value of the parameter
+
+        Example with `zfit`:
+            >>> mean = zfit.Parameter("mean")
+            >>> sampler = calc.sampler(floating_params=[mean])
+            >>> sample = calc.sample(sampler, 1000, mean, 1.2)
+        """
+        return self._sample(sampler, ntoys, parameter, value)
+
+    def toys_loss(self, parameter):
+        """
+        Construct a loss function constructed with a sampler for a given floating parameter
+
+        Args:
+            parameter: floating parameter in the sampler
+        Returns:
+             Loss function
+
+        Example with `zfit`:
+            >>> loss = calc.toys_loss(zfit.Parameter("mean"))
```

Comment:
Why would the user have to pass a parameter object? Would it not be safer and cleaner to provide the name? In fact, how can you connect an object you have just defined (`zfit.Parameter("mean")` in your example) with the parameter stored in the model behind the scenes :S?

Reply:
You have a point you gave me idea of a check for the existence of the parameter in the loss/model. Regarding providing the name I am not sure if it is safer than providing the parameter itself. The name is a property of the parameter.

Reply:
Yes and yes :-).

Reminds me: I saw no `try` anywhere but there may be places where it would be safer to allow for a mistake, and catch it. I have no specific place to add such thing right now, and would have to browse it all.